### ISSUE TITLE
Error when run the code for toy data

### ISSUE THREAD

**Issue by xishazgh** [lmthang/nmt.matlab#1]
Hi everyone,  
I am new on neural networks . Now I am trying to but I get this error:

---

 trainLSTM('../output/id.1000/train.10k', '../output/id.1000/valid.100', '../output/id.1000/test.100', 'de', 'en', '../output/id.1000/train.10k.de.vocab.1000', '../output/id.1000/train.10k.en.vocab.1000', '../output/basic', 'isResume', 0)

 Loading vocab from file ../output/id.1000/train.10k.en.vocab.1000 ...
 Loading vocab from file ../output/id.1000/train.10k.en.vocab.1000 ...
 Loading vocab from file ../output/id.1000/train.10k.de.vocab.1000 ...
Bilingual setting
Init LSTM parameters using dataType=double, initRange=0.1
  Model size = 458700, individual sizes:  W_src{1}=80000 W_tgt{1}=80000 W_emb_src=99500 W_emb_tgt=99600 W_soft=99600
 assert = 0
 attnFunc = 0
 attnOpt = 0
 batchSize = 128
 dataType = double
 debug = 0
 decode = 1
 dropout = 1
 epochFraction = 1
 feedInput = 0
 finetuneEpoch = 5
 finetuneRate = 0.5
 gpuDevice = 0
 initRange = 0.1
 isBi = 1
 isClip = 1
 isGradCheck = 0
 isProfile = 0
 isResume = 0
 isReverse = 0
 learningRate = 1
 loadModel = 
 logFreq = 10
 lstmOpt = 0
 lstmSize = 100
 maxGradNorm = 5
 maxLenRatio = 1.5
 maxSentLen = 51
 minLenRatio = 0.5
 numEpoches = 10
 numLayers = 1
 onlyCPU = 0
 outDir = ../output/basic
 posWin = 10
 saveHDF = 0
 seed = 0
 shuffle = 1
 sortBatch = 1
 srcLang = de
 srcVocabFile = ../output/id.1000/train.10k.de.vocab.1000
 testPrefix = ../output/id.1000/test.100
 tgtLang = en
 tgtVocabFile = ../output/id.1000/train.10k.en.vocab.1000
 trainPrefix = ../output/id.1000/train.10k
 validPrefix = ../output/id.1000/valid.100
 chunkSize = 12800
 baseIndex = 0
 clipForward = 50
 clipBackward = 1000
 nonlinear_gate_f = sigmoid
 nonlinear_gate_f_prime = sigmoidPrime
 nonlinear_f = tanh
 nonlinear_f_prime = tanhPrime
 beamSize = 12
 stackSize = 100
 unkPenalty = 0
 lengthReward = 0
 isGPU = 0
 logId = 6
 srcSos = 994
 srcEos = 995
 srcVocabSize = 995
 nullPosId = 0
 tgtSos = 995
 tgtEos = 996
 tgtVocabSize = 996
 modelFile = ../output/basic/model.mat
 modelRecentFile = ../output/basic/modelRecent.mat
 softmaxSize = 100
 lr = 1
 epoch = 1
 bestCostValid = 100000
 testPerplexity = 100000
 curTestPerpWord = 100000
 startIter = 0
 iter = 0
 epochBatchCount = 0
 finetuneCount = 0
 modelSize = 458700
 Loading data src from file ../output/id.1000/valid.100.de
  src 1:gegen <unk> <unk> , um der <unk> von <unk> <unk>
  src end:<unk> eine <unk> die <unk> des <unk> von <unk> <unk> hat , <unk> eine for <unk> eine <unk> .
 Loading data tgt from file ../output/id.1000/valid.100.en
  tgt 1:than <unk> <unk> to <unk> the <unk> ##AT##-##AT## <unk> of <unk>
  tgt end:<unk> one <unk> <unk> a <unk> in the <unk> of <unk> <unk> , however <unk> an <unk> .
  numSents=100, numWords=1953
 Loading data src from file ../output/id.1000/test.100.de
  src 1:<unk> <unk> und <unk> <unk> <unk> sich gibt Verf眉gung
  src end:Der <unk> aber ein <unk> des <unk> <unk> <unk> vor und sie <unk> am <unk> , dass die <unk> des <unk> stehen <unk> <unk> und <unk> per <unk> 3 <unk> <unk> , um <unk> , Ort es f眉r <unk> <unk> <unk> auf <unk> <unk> <unk> , 2 er <unk> <unk> <unk> .
 Loading data tgt from file ../output/id.1000/test.100.en
  tgt 1:<unk> <unk> and <unk> <unk> minute <unk> land time
  tgt end:The <unk> <unk> a <unk> of the <unk> beginning and <unk> <unk> <unk> that the <unk> &apos;s double <unk> <unk> &quot; <unk> <unk> and <unk> in <unk> <unk> &quot; to <unk> mine the <unk> <unk> guests en <unk> <unk> &apos;s <unk> <unk> <unk> .
  numSents=100, numWords=2225
 Load train data srcFile "../output/id.1000/train.10k.de" and tgtFile "../output/id.1000/train.10k.en"
  src 1:<unk> Unsere aus der Ihrer <unk> .
  src end:wei脽 <unk> Ein ich <unk> und auch <unk> <unk> eingesetzt <unk> sei , wie <unk> es Modul war : Express <unk> einen <unk> , New dieser als <unk> Wann 17 <unk> , der auch gibt from <unk> war , ihm <unk> <unk> von seiner zu seiner durch ihr <unk> gehen zu <unk> , f眉r eine so <unk> <unk> , da脽 da die <unk> der <unk> am <unk> <unk> <unk> <unk> <unk> , und guten auf eine Apartments , da脽 sich in Dateien eine <unk> <unk> <unk> und <unk> <unk> .
  tgt:<unk> <unk> from 鈥?do <unk> .
  tgt end:cities shall <unk> We <unk> <unk> apartment , and restaurants has 24 , &apos;t were <unk> it was that an <unk> a <unk> , of a mobile first <unk> de of <unk> , and one , bit , we was <unk> to the <unk> of <unk> a <unk> <unk> by his entire unto <unk> , Lord be <unk> a <unk> of <unk> <unk> to <unk> the <unk> of the some <unk> of the available download <unk> of the information , and in a <unk> to having in about a <unk> of the available <unk> 鈥?do <unk> and reviews <unk> .
  src input 1: <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> Teil Entwicklung N盲he Sterne des <unk> Hotel in <unk> <unk> kommen Sie eine dir .
  src mask: 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1
  tgt input 1: <t_sos> than ##UNDERSCORE## Update is 00 to short your % at <unk> Hotel in <unk> <unk> . <t_eos>
  tgt output 1: than ##UNDERSCORE## Update is 00 to short your % at <unk> Hotel in <unk> <unk> . <t_eos> <t_eos>
  tgt mask: 1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  0
 Epoch 1, lr=1, 30-Oct-2015 19:47:17
 varsDenseUpdate: W_src W_tgt W_soft
Index exceeds matrix dimensions.

Error in lstmCostGrad (line 110)
          x_t = W_emb(:, input(:, tt));

Error in trainLSTM (line 269)
      [costs, grad] = lstmCostGrad(model, trainData, params, 0);
Could anyone help me with it?  Thanks

**Comment by lmthang** [comment]
Hi,

Can you verify if the step to preprocess data run correctly on the toy data
provided? Why does your output has Chinese characters in it?

EDU>> trainLSTM('../output/id.1000/train.10k',
'../output/id.1000/valid.100', '../output/id.1000/test.100', 'de', 'en',
'../output/id.1000/train.10k.de.vocab.1000',
'../output/id.1000/train.10k.en.vocab.1000', '../output/basic', 'isResume',
0)

# Loading vocab from file ../output/id.1000/train.10k.en.vocab.1000 ...

# Loading vocab from file ../output/id.1000/train.10k.de.vocab.1000 ...

## Bilingual setting

# Init LSTM parameters using dataType=double, initRange=0.1

  Model size = 460600, individual sizes:  W_src{1}=80000 W_tgt{1}=80000
W_emb_src=100200 W_emb_tgt=100200 W_soft=100200

# assert = 0

# attnFunc = 0

# attnOpt = 0

# batchSize = 128

# dataType = double

# debug = 0

# decode = 1

# dropout = 1

# epochFraction = 1

# epochIter = 0

# feedInput = 0

# finetuneEpoch = 5

# finetuneRate = 0.5

# gpuDevice = 0

# initRange = 0.1

# isBi = 1

# isClip = 1

# isGradCheck = 0

# isProfile = 0

# isResume = 0

# isReverse = 0

# learningRate = 1

# loadModel =

# logFreq = 10

# lstmOpt = 0

# lstmSize = 100

# maxGradNorm = 5

# maxLenRatio = 1.5

# maxSentLen = 51

# minLenRatio = 0.5

# numEpoches = 10

# numLayers = 1

# onlyCPU = 0

# outDir = ../output/basic

# posWin = 10

# saveHDF = 0

# seed = 0

# shuffle = 1

# sortBatch = 1

# srcLang = de

# srcVocabFile = ../output/id.1000/train.10k.de.vocab.1000

# testPrefix = ../output/id.1000/test.100

# tgtLang = en

# tgtVocabFile = ../output/id.1000/train.10k.en.vocab.1000

# trainPrefix = ../output/id.1000/train.10k

# validPrefix = ../output/id.1000/valid.100

# chunkSize = 12800

# baseIndex = 0

# clipForward = 50

# clipBackward = 1000

# nonlinear_gate_f = sigmoid

# nonlinear_gate_f_prime = sigmoidPrime

# nonlinear_f = tanh

# nonlinear_f_prime = tanhPrime

# beamSize = 12

# stackSize = 100

# unkPenalty = 0

# lengthReward = 0

# isGPU = 0

# logId = 3

# srcSos = 1001

# srcEos = 1002

# srcVocabSize = 1002

# nullPosId = 0

# tgtSos = 1001

# tgtEos = 1002

# tgtVocabSize = 1002

# modelFile = ../output/basic/model.mat

# modelRecentFile = ../output/basic/modelRecent.mat

# softmaxSize = 100

# lr = 1

# epoch = 1

# bestCostValid = 100000

# testPerplexity = 100000

# curTestPerpWord = 100000

# startIter = 0

# iter = 0

# epochBatchCount = 0

# finetuneCount = 0

# modelSize = 460600

# Loading data src from file ../output/id.1000/valid.100.de

  src 1:Eine <unk> <unk> , um der <unk> von <unk> <unk>
  src end:<unk> eine <unk> die <unk> des <unk> von <unk> <unk> hat , <unk>
eine andere <unk> eine <unk> .

# Loading data tgt from file ../output/id.1000/valid.100.en

  tgt 1:A <unk> <unk> to <unk> the <unk> ##AT##-##AT## <unk> of <unk>
  tgt end:<unk> one <unk> <unk> a <unk> in the <unk> of <unk> <unk> ,
another <unk> an <unk> .
  numSents=100, numWords=1953

# Loading data src from file ../output/id.1000/test.100.de

  src 1:<unk> <unk> und <unk> <unk> <unk> sich noch immer
  src end:Der <unk> liegt ein <unk> des <unk> <unk> <unk> vor und sie <unk>
am <unk> , dass die <unk> des <unk> â€ž <unk> <unk> und <unk> Ende <unk>
â€œ <unk> <unk> , um <unk> , ob es fÃ¼r <unk> <unk> <unk> auf <unk> <unk>
<unk> , wenn er <unk> <unk> <unk> .

# Loading data tgt from file ../output/id.1000/test.100.en

  tgt 1:<unk> <unk> and <unk> <unk> still <unk> each other
  tgt end:The <unk> <unk> a <unk> of the <unk> book and <unk> <unk> <unk>
that the <unk> &apos;s top <unk> <unk> &quot; <unk> <unk> and <unk> in
<unk> <unk> &quot; to <unk> whether the <unk> <unk> could help <unk> <unk>
&apos;s <unk> <unk> <unk> .
  numSents=100, numWords=2225

# Load train data srcFile "../output/id.1000/train.10k.de" and tgtFile

"../output/id.1000/train.10k.en"
  src 1:<unk> dich aus der c <unk> .
  src end:23 <unk> habe ich <unk> und auch <unk> <unk> darÃ¼ber <unk>
mÃ¼ssen , wie <unk> es doch war : Man <unk> einen <unk> , wenig mehr als
<unk> Jahre alten <unk> , der auch noch dazu <unk> war , seinen <unk> <unk>
von Tag zu Tag durch seine <unk> Arbeit zu <unk> , fÃ¼r eine so <unk> <unk>
, daÃŸ ihm die <unk> der <unk> am <unk> <unk> <unk> <unk> <unk> , und zwar
auf eine Weise , daÃŸ sich in ihnen eine <unk> <unk> <unk> und <unk> <unk> .
  tgt:<unk> <unk> from c <unk> .
  tgt end:23 It <unk> me <unk> <unk> then , and often has since , how very
<unk> it was that an <unk> a <unk> , of a little over <unk> years of <unk>
, and one , too , who was <unk> to the <unk> of <unk> a <unk> <unk> by his
daily b <unk> , should be <unk> a <unk> of <unk> <unk> to <unk> the <unk>
of the great <unk> of the most popular <unk> of the day , and in a <unk> to
create in them a <unk> of the most <unk> c <unk> and d <unk> .
  src input 1: <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos>
<s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos>
<s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos>
<s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <s_sos> <unk> , auf
<unk> <unk> <unk> , ist eine kleine <unk> in der <unk> <unk> im ...
  src mask: 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
 0  0  0  0  0  0  0  0  0  0  0  0  1  1  1  1  1  1  1  1  1  1  1  1  1
 1  1  1  1  1
  tgt input 1: <t_sos> <unk> also known as <unk> in <unk> is a small town
in the <unk> of <unk> of <unk> ...
  tgt output 1: <unk> also known as <unk> in <unk> is a small town in the
<unk> of <unk> of <unk> ... <t_eos>
  tgt mask: 1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1

# Epoch 1, lr=1, 04-Nov-2015 01:52:44

# varsDenseUpdate: W_src W_tgt W_soft

**Comment by xishazgh** [comment]
Hi lmthang ,

  I found the cause of that error is not code bug but the version of matlab. 

 Before I used the version is matlab R2013a, now I am using matlab R2015b. It work !

 Thank you very much for your help.

Regards,

Guanhong
